samplePDF/samplePDFFDBase.cpp

```diff
@@ -1414,11 +1358,103 @@ void samplePDFFDBase::addData(TH2D* Data) {
   }
 }
 
+void samplePDFFDBase::SetupNuOscillator() {
+  OscProbCalcerFactory* OscProbCalcFactory = new OscProbCalcerFactory();
+  OscillatorFactory* OscillFactory = new OscillatorFactory();
+  
+  NuOscProbCalcers = std::vector<OscillatorBase*>((int)MCSamples.size());
+  for (int iSample=0;iSample<(int)MCSamples.size();iSample++) {
```

Comment:
NuOscProbCalcers is not deleted in destructor :(

Reply:
this is now done https://github.com/mach3-software/MaCh3/blob/feature_OscClassMagic/samplePDF/samplePDFFDBase.cpp#L46